src/Umbraco.AuthorizedServices/Services/Implement/AuthorizedRequestBuilder.cs

```diff
@@ -33,6 +35,37 @@ internal sealed class AuthorizedRequestBuilder : IAuthorizedRequestBuilder
         return requestMessage;
     }
 
+    public HttpRequestMessage CreateRequestMessage<TRequest>(
```

Comment:
You could perhaps clean up the implementation here if, instead of providing `ServiceDetail` as a parameter, pass a non-null `ApiKeyProvision` instance and the non-null `string apiKey`.

Reply:
The method required the `ApiHost` property from `ServiceDetail` as well as the `GetRequestContent` method. Changing would create a cascade of other changes.

Reply:
That's a fair point.  I've made a small update that at least cleans up the multiple null checks by doing some guards before the method runs.

I'm thinking this might still change if and when we introduce the feature to allow manual entry of API keys via the backoffice.  In that case, the API key won't be in configuration, and we'll probably want to pass it into the method from wherever it has been sourced.  But we can think again when we get to that.